cmake/HPX_AddExecutable.cmake

```diff
@@ -140,7 +140,7 @@ function(add_hpx_executable name)
     endforeach()
   endif()
 
-  if(HPX_WITH_CUDA AND NOT HPX_WITH_CUDA_CLANG)
+  if((HPX_WITH_CUDA OR HPX_WITH_CUDA_SUPPORT) AND NOT HPX_WITH_CUDA_CLANG)
```

Comment:
What's the difference between `HPX_WITH_CUDA` and `HPX_WITH_CUDA_SUPPORT` ?

Reply:
> What's the difference between HPX_WITH_CUDA and HPX_WITH_CUDA_SUPPORT ?

The` difference is that `HPX_WITH_CUDA` implies full `hpx::compute::cuda` currently, whereas the new CUDA_SUPPORT is a much smaller subset that does not pull in `hpx::compute::cuda` and instead only supplies user facing converters to get a future from a stream or stream event. `HPX_WITH_CUDA_SUPPORT`. `HPX_WITH_CUDA` requires that hpx be compiled with the same compiler restrictions that apply to cuda code (such as c++!4 currently), whereas basic support for futures does not.